### PR TITLE
Improvment #90/review datatypes in pandas

### LINKED ISSUE #90: Review datatypes in pandas.
To increase speed of simulation and decrease the usage of memory it is necessary to review and change the dataformat in dataframes if possible.

### PR DESCRIPTION
**Pull request by JakobWeg**
improves performance and memory usage

### CODE REVIEW

File: simbev/car.py [hunk 1]
```diff
@@ -876,10 +888,13 @@ def export(self, region_directory, simbev):
...
 
                 # fit first row event to start at time step 0
-                activity.at[activity.index[0], "event_start"] = 0
-                activity.at[activity.index[0], "event_time"] = post_event_len
+                activity.at[activity.index[0], "event_start"] = np.int32(0)
+                activity.at[activity.index[0], "event_time"] = np.int32(post_event_len)
                 activity.at[activity.index[0], "timestamp"] = simbev.start_date_output
 
+                activity["event_start"] = activity["event_start"].astype("int32")
+                activity["event_time"] = activity["event_time"].astype("int32")
```
Comment (mosc5) [inline simbev/car.py]: Aren't the last 2 lines enough? What exactly does the casting to int32 on assignment do?

**Review by mosc5 — APPROVED** [review]
Do these few changes actually change all relevant data types? It might be useful to already give correct typing when changing the car output to a dataframe (like using category for the use cases). For calculation the most relevant DataFrame is probably the grid one, so if that is fully converted it might be enough

File: simbev/car.py (hunk 1 above)
Comment (JakobWeg) [reply simbev/car.py]: Thats right. After the last Commit the casting is not done anymore, because the datatype gets casted on assignement already.

**Comment by JakobWeg** [comment]
The Improvment in datattypes decreased the time of simulation in default scenario from 32 to 26 seconds.

**Pull request merged**